west.yml: hal_stm32: update to fix compiling with clang and pin updates

tpambor opened the pull request:
Update hal_stm32 to include fix for compiling for STM32H5 with clang.

Fixes warning about extraneous parentheses when compiling with clang.

```
/home/user/west_workspace/modules/hal/stm32/stm32cube/stm32h5xx/drivers/src/stm32h5xx_hal_rcc_ex.c:3513:26: error: equality comparison with extraneous parentheses [-Werror,-Wparentheses-equality]
 3513 |         else if ((srcclk == RCC_USART2CLKSOURCE_PLL3Q))
      |                   ~~~~~~~^~~~~~~~~~~~~~~~~~~~~~~~~~~~
/home/user/west_workspace/modules/hal/stm32/stm32cube/stm32h5xx/drivers/src/stm32h5xx_hal_rcc_ex.c:3513:26: note: remove extraneous parentheses around the comparison to silence this warning
 3513 |         else if ((srcclk == RCC_USART2CLKSOURCE_PLL3Q))
      |                  ~       ^                           ~
/home/user/west_workspace/modules/hal/stm32/stm32cube/stm32h5xx/drivers/src/stm32h5xx_hal_rcc_ex.c:3513:26: note: use '=' to turn this equality comparison into an assignment
 3513 |         else if ((srcclk == RCC_USART2CLKSOURCE_PLL3Q))
      |                          ^~
      |                          =
```

EDIT (@erwango): Also introduces ALE and CLE pins for the FMC peripheral from a distinct HAL PR.

erwango reviewed (dismissed):
Need to wait #97126 to be merged first

erwango commented:
Btw, if you don't mind, it would be nice to take https://github.com/zephyrproject-rtos/hal_stm32/pull/318 as well. Up to you

tpambor committed:
west.yml: hal_stm32: update to fix compiling with clang and pin updates

Update hal_stm32 to include
- fix for compiling for STM32H5 with clang
- ale/cle pin definitions for FMC peripheral

Signed-off-by: Tim Pambor <tim.pambor@codewrights.de>
